### ISSUE TITLE
fluent bit randomly exit with code 136

### ISSUE THREAD

**Issue by furstenbergs** [fluent/fluent-bit#4035]
## Bug Report

**Describe the bug**

I've deployed fluent-bit via the official helm cart to a K8s cluster running on EKS. At completely random times I see that one of more fluent-bit pods will go into an error state or CrashLoopBackOff. Looking at the pod logs I do not see anything that provides more information around why this happens. When I describe the pod, I see the pod exited with error 136.

```Containers:
  fluent-bit:
    Container ID:   docker://7f82f7f659fd587ffd2e5c4f081cc2e46993e7ee67a0b95772cd6093dd83c01c
    Image:          fluent/fluent-bit:1.8.3
    Image ID:       docker-pullable://fluent/fluent-bit@sha256:10ea2709cef6e7059d980b4969d5f9d753ef97278a817c214cbe9120b1152082
    Port:           2020/TCP
    Host Port:      0/TCP
    State:          Waiting
      Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       Error
      Exit Code:    136
      
```
**To Reproduce**
- No error that I can pick up from the logs, not even with log level set to debug. I only see this on k8s.

```
Warning  BackOff  64s (x709 over 160m)  kubelet, ip-172-26-73-3.eu-west-1.compute.internal  Back-off restarting failed container
```
**Steps to reproduce the problem**
- I've deleted and redeployed a few times. No specific steps to reproduce as this happens at random

**Expected behavior**

Provide information in the logs around why it existed or fail. 

**Screenshots**

**Your Environment**

* Version used: 1.8.3
* Configuration:
* Environment name and version (e.g. Kubernetes? What version?): Kubernetes v1.19.13
* Server type and version: AWS EKS
* Operating System and version: Amazon Linux 2
* Filters and plugins: Kubernetes filter

**Additional context**

Can you please confirm what exit code 136 means?

If I delete the pod and allow the daemonset to create a new one, it typically works fine, until other pods run into the same error

**Comment by paulden** [comment]
I had exactly the same issue with version 1.8.3 where one or two pods per cluster were randomly restarting with exit code 136. Other pods (about 10 per cluster) would be fine though.

I found no useful log in the failing fluent-bit containers, I tried bumping the version to 1.8.5 and so far I have not seen the issue coming back but I'm monitoring them closely.

**Comment by kamaleshtangudu** [comment]
Happening to me on AWS Fargate containers running `amazon/aws-for-fluent-bit:2.19.0`. They are  going down with `exitCode=136`, the image is internally using `fluent/fluent-bit:1.8.3` . It goes down without any error in the logs. Need to know what the exit code 136 means? Need logs when this error occurs.

**Comment by furstenbergs** [comment]
I also upgraded to version 1.8.5 as mentioned by @paulden and for the past 18 hours, all pods have been running fine. 

I Will keep a close eye on them but seems to be fixed in version 1.8.5

**Comment by nokute78** [comment]
Are you using elasticsearch plugin ?
If so, #3905 may be the same issue.

I sent a patch #3931 and it was merged to v1.8.4.
https://fluentbit.io/announcements/v1.8.4/

**Comment by furstenbergs** [comment]
Yes, using elasticsearch plugin.

Since upgrading to v1.8.5 I have not seen this issue again. Closing this for now.
